File: Reverse Words in a String/Reverse_Words_in_a_String.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        string temp;
        vector<string> arr;
        while(ss>>temp) {
            arr.push_back(temp);
        }
        reverse(arr.begin(),arr.end());
        string ans = "";
        for(auto i: arr) {
            ans += i+" ";
        }
        ans.pop_back();
        return ans;
    }
};

File: Reverse Words in a String/Reverse_Words_in_a_String.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        String arr[] = s.split("\\s+");
        int left = 0,right = arr.length-1;
        while(left < right) {
            String temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
        StringBuilder sb = new StringBuilder("");
        for(int i = 0; i<arr.length; i++) {
            sb.append(arr[i]);
            if(i != arr.length-1) sb.append(" ");
        }
        return sb.toString();
    }
}